File: app/src/main/java/com/wordpress/lonelytripblog/funwithflags/di/FragmentsContributor.java
package com.wordpress.lonelytripblog.funwithflags.di;

import com.wordpress.lonelytripblog.funwithflags.ui.AboutFragment;
import com.wordpress.lonelytripblog.funwithflags.ui.GameFragment;
import com.wordpress.lonelytripblog.funwithflags.ui.HomeFragment;
import com.wordpress.lonelytripblog.funwithflags.ui.InfoFragment;
import com.wordpress.lonelytripblog.funwithflags.ui.RecapFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class FragmentsContributor {

    @ContributesAndroidInjector
    abstract HomeFragment homeFragment();

    @ContributesAndroidInjector
    abstract GameFragment gameFragment();

    @ContributesAndroidInjector
    abstract RecapFragment recapFragment();

    @ContributesAndroidInjector
    abstract InfoFragment infoFragment();

    @ContributesAndroidInjector
    abstract AboutFragment aboutFragment();

}

File: app/src/main/java/com/wordpress/lonelytripblog/funwithflags/di/FragmentsContributor.kt
package com.wordpress.lonelytripblog.funwithflags.di

import com.wordpress.lonelytripblog.funwithflags.ui.*
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class FragmentsContributor {
    @ContributesAndroidInjector
    abstract fun contributeHomeFragment(): HomeFragment

    @ContributesAndroidInjector
    abstract fun contributeGameFragment(): GameFragment

    @ContributesAndroidInjector
    abstract fun contributeRecapFragment(): RecapFragment

    @ContributesAndroidInjector
    abstract fun contributeAboutFragment(): AboutFragment
}